src/main/webapp/app/exercises/programming/manage/update/update-components/language/programming-exercise-language.component.html

```diff
@@ -0,0 +1,240 @@
+                />
+                @for (e of packageName.errors! | keyvalue | removekeys: ['required']; track e) {
+                    @if (packageName.invalid && (packageName.dirty || packageName.touched)) {
+                        <div class="alert alert-danger">
+                            @if (programmingExercise.projectType === ProjectType.MAVEN_BLACKBOX) {
+                                <div [jhiTranslate]="'artemisApp.exercise.form.packageName' + '.' + e.key + '.' + programmingExercise.programmingLanguage + '_BLACKBOX'"></div>
+                            }
+                            @if (programmingExercise.projectType !== ProjectType.MAVEN_BLACKBOX) {
+                                <div [jhiTranslate]="'artemisApp.exercise.form.packageName' + '.' + e.key + '.' + programmingExercise.programmingLanguage"></div>
+                            }
+                        </div>
+                    }
+                }
+            </div>
+        }
+        @if (programmingExercise.programmingLanguage === ProgrammingLanguage.SWIFT && programmingExercise.projectType === ProjectType.XCODE) {
+            <div class="form-group">
+                <label class="label-narrow" jhiTranslate="artemisApp.programmingExercise.appName" for="field_packageName"></label>
+                <input
+                    required
+                    type="text"
+                    [pattern]="programmingExerciseCreationConfig.appNamePatternForSwift"
+                    class="form-control"
+                    name="packageName"
+                    id="field_appName"
+                    [(ngModel)]="programmingExercise.packageName"
+                    [readonly]="programmingExercise.id"
+                    #packageName="ngModel"
+                    [disabled]="programmingExerciseCreationConfig.isImportFromExistingExercise || !!programmingExercise.id || programmingExerciseCreationConfig.isImportFromFile"
+                />
+                @for (e of packageName.errors! | keyvalue | removekeys: ['required']; track e) {
+                    @if (packageName.invalid && (packageName.dirty || packageName.touched)) {
+                        <div class="alert alert-danger">
+                            <div [jhiTranslate]="'artemisApp.exercise.form.packageName' + '.' + e.key + '.' + programmingExercise.programmingLanguage"></div>
+                        </div>
+                    }
+                }
+            </div>
+        }
+    }
+
+    <!-- Theia Image Selection -->
+    @if (programmingExercise.allowOnlineIde && programmingExercise.programmingLanguage) {
+        <label class="label-narrow" jhiTranslate="artemisApp.programmingExercise.theiaImage.title"></label>
+        <jhi-programming-exercise-theia [programmingExercise]="programmingExercise" [programmingExerciseCreationConfig]="programmingExerciseCreationConfig" />
+    }
```

Comment:
TODO add option to hide / dispay theia

Reply:
Field is disabled where Theia can be set to true, so no additional check should be needed here